tasks/main.yml

```diff
@@ -1,186 +1,33 @@
-    - cartridge-instances
-    - cartridge-replicasets
-    - cartridge-config
 
-- name: Check distribution
-  fail:
-    msg: 'Deploy to {{ ansible_os_family }} distributions is not supported yet'
-  when: ansible_os_family not in ["RedHat", "Debian"]
-  tags:
-    - cartridge-instances
-    - cartridge-replicasets
-    - cartridge-config
-
-- name: Set remote_user for delegated tasks
-  set_fact:
-    remote_user: '{{ ansible_user }}'
-  when: remote_user is not defined and ansible_user is defined
+- name: 'Forced gathering facts'
+  setup:
+  when: module_setup is not defined
   tags:
     - cartridge-instances
     - cartridge-replicasets
     - cartridge-config
 
-- name: Validate config
-  cartridge_validate_config:
-    hosts: '{{ play_hosts }}'
-    hostvars: '{{ hostvars }}'
-  delegate_to: localhost
-  become: false
-  run_once: true
+- name: 'Validate config and hosts'
+  import_tasks: 'validate.yml'
   tags:
     - cartridge-instances
     - cartridge-replicasets
     - cartridge-config
 
-- name: Collect instance info
-  cartridge_set_instance_info:
-    app_name: '{{ cartridge_app_name }}'
-    instance_name: '{{ inventory_hostname }}'
-    instance_vars:
-      cartridge_package_path: '{{ cartridge_package_path }}'
-      cartridge_dist_dir: '{{ cartridge_dist_dir }}'
-      cartridge_conf_dir: '{{ cartridge_conf_dir }}'
-      cartridge_run_dir: '{{ cartridge_run_dir }}'
-      cartridge_data_dir: '{{ cartridge_data_dir }}'
-      stateboard: '{{ stateboard }}'
+- name: 'Prepare facts'
+  import_tasks: 'prepare.yml'
   tags:
     - cartridge-instances
     - cartridge-replicasets
     - cartridge-config
 
-- name: Get one instance from each physical machine
-  cartridge_set_single_instances_for_each_machine:
-    hostvars: '{{ hostvars }}'
-    play_hosts: '{{ play_hosts }}'
-  run_once: true
-  tags: cartridge-instances
-
-- name: Install package
-  include_tasks: install_package.yml
-  when: >-
-    cartridge_package_path is not none and
-    inventory_hostname in single_instances_for_each_machine
-  tags: cartridge-instances
-
-- name: Start instance
-  include_tasks: start_instance.yml
-  when: not expelled
-  tags: cartridge-instances
-
-- name: Restart instances and reload systemd-daemon
-  meta: flush_handlers
-  tags: cartridge-instances
-
-- name: Wait for instance to be started
-  cartridge_check_instance_state:
-    console_sock: '{{ instance_info.console_sock }}'
-    stateboard: '{{ stateboard }}'
-  register: check_instance
-  until: not check_instance.failed
-  retries: '{{ instance_start_timeout // 5 }}'
-  delay: 5
-  when: not expelled
-  tags: cartridge-instances
-
-- name: Connect instance to membership
-  cartridge_probe_instance:
-    console_sock: '{{ instance_info.console_sock }}'
-    hostvars: '{{ hostvars }}'
-    play_hosts: '{{ play_hosts }}'
-  register: probe
-  until: not probe.failed
-  retries: 3
-  delay: 5
-  when: not expelled and not stateboard
+- name: 'BLOCK : Import tasks by scenario'
```

Comment:
Maybe add some filter to format block names? Like `name: '{{ "Import tasks by scenario" | cartridge_block }}' `.

Reply:
It seems to me that it will be less readable

Reply:
But it removes a copy-paste. What about 

`name: '{{ "Import tasks by scenario" | format_cartridge_block_name }}'`

Reply:
No, you will use copy-paste `format_cartridge_block_name`)